### PR TITLE
Use gettimeofday() in perftest and if rdtsc is unstable

### LINKED ISSUE #2792: Frequency read by UCS on x86 doesn't match actual CPU frequency
`ucs_arch_get_clocks_per_sec()` in `src/ucs/arch/x86_64/cpu.c` reads the frequency using the CPU model instead of information from cpuinfo. This is incorrect when the user changes the frequency of the CPU to something other than the base frequency of the CPU. The correct CPU frequency will be read only from `ucs_get_cpuinfo_clock_freq` when the CPU frequency is different from the base frequency.

### PR DESCRIPTION
**Pull request by yosefe**
Replaces #2793 
Fixes #2792
@rzambre 

This PR also removes the CPU frequency warning. Instead of warning, just fall back to gettimeofday().
Currently ucs_get_time() is not used in data path. Even UD transport samples the timer during async thread only.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8366/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8371/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov @hoopoepg @brminich can you pls take a look?

File: src/ucs/arch/x86_64/cpu.h [hunk 1]
```diff
@@ -37,9 +37,15 @@ BEGIN_C_DECLS
 #define ucs_memory_cpu_load_fence()   ucs_compiler_fence()
 #define ucs_memory_cpu_wc_fence()     asm volatile ("sfence" ::: "memory")
 
+extern int ucs_arch_x86_enable_rdtsc;
+
 
 static inline uint64_t ucs_arch_read_hres_clock()
 {
+    if (ucs_unlikely(!ucs_arch_x86_enable_rdtsc)) {
```
Comment (hoopoepg) [inline src/ucs/arch/x86_64/cpu.h]: minor: declaration first

Comment (yosefe) [reply src/ucs/arch/x86_64/cpu.h]: which declaration do you mean?

Comment (hoopoepg) [reply src/ucs/arch/x86_64/cpu.h]: on line 49:
```
uint32_t low, high;
```

File: src/tools/perf/libperf.c [hunk 2]
```diff
@@ -225,9 +225,15 @@ static void uct_perf_test_free_mem(ucx_perf_context_t *perf)
 
 void ucx_perf_test_start_clock(ucx_perf_context_t *perf)
 {
-    perf->start_time        = ucs_get_time();
-    perf->prev_time         = perf->start_time;
-    perf->prev.time         = perf->start_time;
+    ucs_time_t start_time = ucs_get_time();
+
+    perf->start_time_acc   = ucs_get_accurate_time();
+    perf->end_time         = (perf->params.max_time == 0.0) ? UINT64_MAX :
```
Comment (evgeny-leksikov) [inline src/tools/perf/libperf.c]: `ucs_time_t                   end_time;`
lets define `UCS_TIME_INF` constant or use smth like `ucs_time_from_sec(UINT64_MAX)`

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8377/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8378/ for details (Mellanox internal link).

**Pull request merged**